google/cloud/storage/testing/object_integration_test.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/testing/object_integration_test.h"
+#include "google/cloud/internal/getenv.h"
+#include "google/cloud/status.h"
+#include <gmock/gmock.h>
+#include <cstddef>
+#include <cstdlib>
+#ifdef __unix__
+#include <dirent.h>
+#endif  // __unix__
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace storage {
+namespace testing {
+namespace {
+
+StatusOr<std::size_t> GetNumEntries(std::string const& path) {
+#ifdef __unix__
+  dirent** namelist;
+  int res = scandir(path.c_str(), &namelist, nullptr, alphasort);
+  if (res < 0) {
+    return Status(StatusCode::kInternal, "Failed to open directory \"" + path +
+                                             "\": " + strerror(errno));
+  }
+  for (int i = 0; i < res; ++i) {
+    free(namelist[i]);
+  }
+  free(namelist);
```

Comment:
It looks like a `readdir()` loop would be simpler here, and avoid extra allocations and unnecessary sorting.